.buildkite/scripts/promote_release_channel.sh

```diff
@@ -74,19 +74,19 @@ fi
 # atomically promote a set of packages, we can simplify all this to
 # just promote everything at once. Until that time, however, please
 # leave this arrangement "as is".
-non_supervisor_packages=($(echo "${channel_pkgs_json}" | \
+mapfile -t non_supervisor_packages < <(echo "${channel_pkgs_json}" | \
```

Comment:
Make sure we do some tests here; this should work *assuming* `channel_pkgs_json` ["outputs multiple lines, each of which should be an element"](https://github.com/koalaman/shellcheck/wiki/SC2207)

Reply:
We will always end up with `{"range_start":0,"range_end":0,"total_count":0,"data":[]}` at the very least, since this is CURLing the API

Reply:
But then that gets passed through a `jq` command, which makes it into one line per package. I think it's good as is, but if the `jq` command were different (e.g., space- or tab-delimited rather than newline-delimited), it could break.

In the end, I think this looks good; I just hadn't dug into the full details of the command before.

Reply:
the `jq` command in this case is only used in this if statement here: https://github.com/habitat-sh/habitat/blob/9b769c16a89363dba483dad0f324ca7d1610897d/.buildkite/scripts/promote_release_channel.sh#L61-L64 it doesn't mutate the actual contents of the curl. either way, I'm being pedantic, I think we're fine here. :)

Reply:
`jq` *is* used in this command. The full command that we're using as input to `mapfile` is
```bash
echo "${channel_pkgs_json}" | \
                           jq -r \
                             '.data |
                             map(select(.name != "hab-sup")) |
                             map(.origin + "/" + .name + "/" + .version + "/" + .release)
                             | .[]'
```
If we try this manually with the `habitat/stable` channel (since it has a more modest number of packages):
```bash
$ channel_pkgs_json=$(curl "https://bldr.habitat.sh/v1/depot/channels/habitat/stable/pkgs")
$ mapfile -t non_supervisor_packages < <(echo "${channel_pkgs_json}" | jq -r
                               '.data |
                               map(select(.name != "hab-sup")) |
                               map(.origin + "/" + .name + "/" + .version + "/" + .release)
                               | .[]')
$ declare -p non_supervisor_packages
declare -a non_supervisor_packages=([0]="habitat/airlock/2012/20180129223853" [1]="habitat/airlock/2037/20180207210710" [2]="habitat/airlock/2431/20180621184431" [3]="habitat/airlock/3082/20190131224338" [4]="habitat/airlock/3094/20190206175057" [5]="habitat/builder-admin-proxy/7012/20180129225745" [6]="habitat/builder-admin/7012/20180129225031" [7]="habitat/builder-api-proxy/7012/20180129225803" [8]="habitat/builder-api-proxy/7037/20180207200149" [9]="habitat/builder-api-proxy/7052/20180208233113" [10]="habitat/builder-api-proxy/7082/20180216002409" [11]="habitat/builder-api-proxy/7129/20180305064742" [12]="habitat/builder-api-proxy/7135/20180306175628" [13]="habitat/builder-api-proxy/7201/20180310002554" [14]="habitat/builder-api-proxy/7222/20180317010951" [15]="habitat/builder-api-proxy/7244/20180321231217" [16]="habitat/builder-api-proxy/7268/20180328202543" [17]="habitat/builder-api-proxy/7286/20180403161615" [18]="habitat/builder-api-proxy/7288/20180410220811" [19]="habitat/builder-api-proxy/7292/20180412181836" [20]="habitat/builder-api-proxy/7316/20180429021438" [21]="habitat/builder-api-proxy/7332/20180504011014" [22]="habitat/builder-api-proxy/7336/20180507163944" [23]="habitat/builder-api-proxy/7346/20180510202318" [24]="habitat/builder-api-proxy/7348/20180521173035" [25]="habitat/builder-api-proxy/7356/20180523175804" [26]="habitat/builder-api-proxy/7411/20180613221149" [27]="habitat/builder-api-proxy/7423/20180620130107" [28]="habitat/builder-api-proxy/7459/20180706233113" [29]="habitat/builder-api-proxy/7486/20180718235151" [30]="habitat/builder-api-proxy/7604/20180906222848" [31]="habitat/builder-api-proxy/7648/20180914184917" [32]="habitat/builder-api-proxy/7652/20180915164017" [33]="habitat/builder-api-proxy/7668/20180915185905" [34]="habitat/builder-api-proxy/7672/20180916051807" [35]="habitat/builder-api-proxy/7698/20180921203254" [36]="habitat/builder-api-proxy/7702/20180921222730" [37]="habitat/builder-api-proxy/7766/20181008185242" [38]="habitat/builder-api-proxy/7780/20181010164756" [39]="habitat/builder-api-proxy/7884/20181102184440" [40]="habitat/builder-api-proxy/7940/20181113205912" [41]="habitat/builder-api-proxy/7975/20181127224434" [42]="habitat/builder-api-proxy/7985/20181130051220" [43]="habitat/builder-api-proxy/8049/20190122233006" [44]="habitat/builder-api-proxy/8097/20190206214259" [45]="habitat/builder-api/7012/20180129230016" [46]="habitat/builder-api/7052/20180208233114" [47]="habitat/builder-api/7127/20180305050823" [48]="habitat/builder-api/7203/20180312011353" [49]="habitat/builder-api/7209/20180314213927")
```
If we removed the `jq`, `non_supervisor_packages` would be wrong (note the single element array):
```bash
$ mapfile -t non_supervisor_packages < <(echo "${channel_pkgs_json}")
$ $ declare -p non_supervisor_packages
declare -a non_supervisor_packages=([0]="{\"range_start\":0,\"range_end\":49,\"total_count\":245,\"data\":[{\"origin\":\"habitat\",\"name\":\"airlock\",\"version\":\"2012\",\"release\":\"20180129223853\"},{\"origin\":\"habitat\",\"name\":\"airlock\",\"version\":\"2037\",\"release\":\"20180207210710\"},{\"origin\":\"habitat\",\"name\":\"airlock\",\"version\":\"2431\",\"release\":\"20180621184431\"},{\"origin\":\"habitat\",\"name\":\"airlock\",\"version\":\"3082\",\"release\":\"20190131224338\"},{\"origin\":\"habitat\",\"name\":\"airlock\",\"version\":\"3094\",\"release\":\"20190206175057\"},{\"origin\":\"habitat\",\"name\":\"builder-admin-proxy\",\"version\":\"7012\",\"release\":\"20180129225745\"},{\"origin\":\"habitat\",\"name\":\"builder-admin\",\"version\":\"7012\",\"release\":\"20180129225031\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7012\",\"release\":\"20180129225803\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7037\",\"release\":\"20180207200149\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7052\",\"release\":\"20180208233113\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7082\",\"release\":\"20180216002409\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7129\",\"release\":\"20180305064742\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7135\",\"release\":\"20180306175628\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7201\",\"release\":\"20180310002554\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7222\",\"release\":\"20180317010951\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7244\",\"release\":\"20180321231217\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7268\",\"release\":\"20180328202543\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7286\",\"release\":\"20180403161615\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7288\",\"release\":\"20180410220811\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7292\",\"release\":\"20180412181836\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7316\",\"release\":\"20180429021438\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7332\",\"release\":\"20180504011014\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7336\",\"release\":\"20180507163944\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7346\",\"release\":\"20180510202318\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7348\",\"release\":\"20180521173035\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7356\",\"release\":\"20180523175804\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7411\",\"release\":\"20180613221149\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7423\",\"release\":\"20180620130107\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7459\",\"release\":\"20180706233113\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7486\",\"release\":\"20180718235151\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7604\",\"release\":\"20180906222848\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7648\",\"release\":\"20180914184917\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7652\",\"release\":\"20180915164017\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7668\",\"release\":\"20180915185905\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7672\",\"release\":\"20180916051807\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7698\",\"release\":\"20180921203254\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7702\",\"release\":\"20180921222730\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7766\",\"release\":\"20181008185242\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7780\",\"release\":\"20181010164756\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7884\",\"release\":\"20181102184440\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7940\",\"release\":\"20181113205912\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7975\",\"release\":\"20181127224434\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"7985\",\"release\":\"20181130051220\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"8049\",\"release\":\"20190122233006\"},{\"origin\":\"habitat\",\"name\":\"builder-api-proxy\",\"version\":\"8097\",\"release\":\"20190206214259\"},{\"origin\":\"habitat\",\"name\":\"builder-api\",\"version\":\"7012\",\"release\":\"20180129230016\"},{\"origin\":\"habitat\",\"name\":\"builder-api\",\"version\":\"7052\",\"release\":\"20180208233114\"},{\"origin\":\"habitat\",\"name\":\"builder-api\",\"version\":\"7127\",\"release\":\"20180305050823\"},{\"origin\":\"habitat\",\"name\":\"builder-api\",\"version\":\"7203\",\"release\":\"20180312011353\"},{\"origin\":\"habitat\",\"name\":\"builder-api\",\"version\":\"7209\",\"release\":\"20180314213927\"}]}")
```